sql/hive/src/main/scala/org/apache/spark/sql/hive/test/TestHive.scala

```diff
@@ -83,26 +154,6 @@ class TestHiveContext(sc: SparkContext) extends HiveContext(sc) {
 
   hiveconf.set("hive.plan.serialization.format", "javaXML")
 
-  lazy val warehousePath = Utils.createTempDir(namePrefix = "warehouse-")
-
-  lazy val scratchDirPath = {
-    val dir = Utils.createTempDir(namePrefix = "scratch-")
-    dir.delete()
-    dir
-  }
-
-  private lazy val temporaryConfig = newTemporaryConfiguration(useInMemoryDerby = false)
-
-  /** Sets up the system initially or after a RESET command */
-  protected override def configure(): Map[String, String] = {
-    super.configure() ++ temporaryConfig ++ Map(
```

Comment:
Looks like as long as we make `temporaryConfig` a constant value, it is fine to have 
```
TestHiveContext.hiveClientConfigurations(hiveconf, warehousePath, scratchDirPath)
  .foreach { case (k, v) => metadataHive.runSqlHive(s"SET $k=$v") }
```
in the reset. I feel reset those confs is still needed just in case we override any of those confs.